build.gradle

```diff
@@ -30,10 +33,26 @@ allprojects {
         mavenCentral()
         google()
         jcenter()
-        
+
+    }
+    tasks.whenTaskAdded { theTask ->
+        if (theTask.name.startsWith('compile')) {
+            theTask.dependsOn ktlintFormat
+        }
     }
 }
 
 task clean(type: Delete) {
     delete rootProject.buildDir
 }
+
+subprojects {
+    apply plugin: "org.jlleitschuh.gradle.ktlint" // Version should be inherited from parent
+
+    // Optionally configure plugin
+    ktlint {
```

Comment:
I think we should lock `ktlint` version here. The version may change between patch versions of `ktlint-gradle`.

Reply:
Locked in c804899